src/Symfony/Component/Console/Tester/CommandTester.php

```diff
@@ -64,6 +68,17 @@ public function execute(array $input, array $options = array())
             $input = array_merge(array('command' => $this->command->getName()), $input);
         }
 
+        if ($this->inputs) {
+            if (null === $this->command->getHelperSet()) {
+                $helper = new QuestionHelper();
+                $this->command->setHelperSet(new HelperSet(array($helper)));
+            } else {
+                $helper = $this->command->getHelper('question');
+            }
+
+            $helper->setInputStream($this->createInputStream());
+        }
```

Comment:
Adding an interface with a `setInputStream` method then implement it in the Input object could be a good solution. We could then automatically use `Input::$inputStream` in every helper that has an `inputStream`, working for both QuestionHelper and SymfonyQuestionHelper. At first looking for the `Input::$inputStream` while letting the `QuestionHelper::$inputStream` as deprecated for finally remove it in 4.0.
What do you think? It would close #18902 and ease interactive command testing properly.